lib/3scale_toolbox/commands/import_command/openapi/mapping_rule.rb

```diff
@@ -17,8 +17,9 @@ def http_method
           end
 
           def pattern
-            # apply strict matching
-            "#{raw_pattern}$"
+            res = "#{raw_pattern}"
+            res = "#{res}$" if !operation[:prefix_matching] # apply strict matching
```

Comment:
I implemented the setting of the pattern depending on the prefix_matching as an attribute of the Operation class.
However I'm not really sure of whether it would be the correct approach depending on what's the design of the separation of concerns among different classes I've seen

Possible solution candidates I've seen are:
 * Send the flag as a parameter of the ThreeScaleApiSpec constructor and also send it as part of the operation hash passed to the Operation class from the "operations" method in ThreeScaleApiSpec as done now.
 * Set the flag as part of the step context and as part of the CreateMappingRule Step modify the returned operations using the set method
 * Set the flag as part of the Swagger.build method. This does not seem really correct as I understand that class should be independent of 3scale specifics

What do you think? Where should the responsibilities be delimited?

Reply:
I agree with the analysis. Options 1) and 2) are the best. Third one does not seem correct.

I would also go for the 1) option